torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -338,6 +338,22 @@ Tensor norm_jvp(
   return norm_jvp(self_p, self_t, p_, norm, {}, true);
 }
 
+Tensor _nested_from_padded_backward(
+    const Tensor& grad,
+    const Tensor& input,
+    const bool do_transform_0213) {
+  if (do_transform_0213) {
+    auto new_sizes = IntArrayRef{
+        input.size(0), input.size(2), (input.size(1) * input.size(3))};
+    auto out = grad.to_padded_tensor(0, new_sizes);
+    auto expand_last_dim_size =
+        IntArrayRef{input.size(0), input.size(2), input.size(1), input.size(3)};
+    out = out.view(expand_last_dim_size).permute({0, 2, 1, 3});
+    return out.reshape(input.sizes());
```

Comment:
why do we need this reshape? shouldn't the shape already be correct after the permute?

Reply:
yeah, good catch